tee_supplicant/src/tee_supplicant.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/tee.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <optee_msg_supplicant.h>
+#include <tee_client_api.h>
+#include <teec_ta_load.h>
+
+LOG_MODULE_REGISTER(tee_supplicant);
+
+#define TEE_SUPP_THREAD_PRIO	7
+
+#define TEE_REQ_PARAM_MAX	5
+
+static struct k_thread main_thread;
+static K_THREAD_STACK_DEFINE(main_stack, 8192);
+
+struct suppl_shm {
+	int id;
+	void *p;
+	size_t size;
+	bool registered;
+	struct suppl_shm *next;
```

Comment:
Any particular reason why zephyr's `dlist` is not used?

Reply:
sparrows from the cannon...

Reply:
I prefer call this "use of well-defined interfaces"

Reply:
This code removed